awx_collection/plugins/modules/tower_group.py

```diff
@@ -104,87 +66,62 @@
         inventory=dict(required=True),
         variables=dict(),
-        credential=dict(),
-        source=dict(choices=["manual", "file", "ec2", "rax", "vmware",
-                             "gce", "azure", "azure_rm", "openstack",
-                             "satellite6", "cloudforms", "custom"]),
-        source_regions=dict(),
-        source_vars=dict(),
-        instance_filters=dict(),
-        group_by=dict(),
-        source_script=dict(),
-        overwrite=dict(type='bool'),
-        overwrite_vars=dict(type='bool'),
-        update_on_launch=dict(type='bool'),
         state=dict(choices=['present', 'absent'], default='present'),
     )
 
+    # Create a module for ourselves
     module = TowerModule(argument_spec=argument_spec, supports_check_mode=True)
 
+    # Extract our parameters
     name = module.params.get('name')
+    new_name = module.params.get('new_name')
     inventory = module.params.get('inventory')
-    credential = module.params.get('credential')
+    description = module.params.get('description')
     state = module.params.pop('state')
-
     variables = module.params.get('variables')
-    if variables:
-        if variables.startswith('@'):
-            filename = os.path.expanduser(variables[1:])
-            with open(filename, 'r') as f:
-                variables = f.read()
-
-    json_output = {'group': name, 'state': state}
-
-    tower_auth = tower_auth_config(module)
-    with settings.runtime_values(**tower_auth):
-        tower_check_mode(module)
-        group = tower_cli.get_resource('group')
-        try:
-            params = module.params.copy()
-            params['create_on_missing'] = True
-            params['variables'] = variables
-
-            inv_res = tower_cli.get_resource('inventory')
-            inv = inv_res.get(name=inventory)
-            params['inventory'] = inv['id']
-
-            if credential:
-                cred_res = tower_cli.get_resource('credential')
-                cred = cred_res.get(name=credential)
-                params['credential'] = cred['id']
-
-            if state == 'present':
-                result = group.modify(**params)
-                json_output['id'] = result['id']
-            elif state == 'absent':
-                result = group.delete(**params)
-        except (exc.NotFound) as excinfo:
-            module.fail_json(msg='Failed to update the group, inventory not found: {0}'.format(excinfo), changed=False)
-        except (exc.ConnectionError, exc.BadRequest, exc.AuthError) as excinfo:
-            module.fail_json(msg='Failed to update the group: {0}'.format(excinfo), changed=False)
-
-    json_output['changed'] = result['changed']
-    module.exit_json(**json_output)
 
+    # Attempt to lookup the related items the user specified (these will fail the module if not found)
+    inventory_id = module.resolve_name_to_id('inventories', inventory)
+
+    # Attempt to lookup team based on the provided name and org ID
```

Comment:
comment wording from copy+paste

I'd prefer to lose the comment completely

Reply:
Good catch, typo fixed for now. It would be nice if we at least got rid of the 'team' in the comment and made it something more generic. My goal with the (excessive?) comments was to provide a clear template to anyone who might want to help us in this endeavor of module creation and maintenance. Since I can't tell which module someone might look at I figured lets just put them in all of them. (otherwise fixed)